### PR TITLE
*: remove refs to deprecated io/ioutil

### PR DESCRIPTION
**Pull request by testwill**
### What problem does this PR solve?

 remove refs to deprecated io/ioutil

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: domain/plan_replayer.go [hunk 1]
```diff
@@ -87,11 +87,17 @@ func (p *dumpFileGcChecker) setupSctx(sctx sessionctx.Context) {
 }
 
 func (p *dumpFileGcChecker) gcDumpFilesByPath(path string, gcDurationDefault, gcDurationForCapture time.Duration) {
-	files, err := ioutil.ReadDir(path)
+	entries, err := os.ReadDir(path)
 	if err != nil {
-		if !os.IsNotExist(err) {
+		logutil.BgLogger().Warn("[dumpFileGcChecker] open plan replayer directory failed", zap.Error(err))
+	}
+	files := make([]fs.FileInfo, 0, len(entries))
```
Comment (tangenta) [inline domain/plan_replayer.go:94]: Maybe we can iterate over `entries` directly?

```
for _, f := range files {
```

=>

```
for _, f := range entries {
```

**Review by tangenta — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**